sklearn/impute.py

```diff
@@ -362,6 +379,26 @@ def _dense_fit(self, X, strategy, missing_values, fill_value):
         # Constant
         elif strategy == "constant":
             return np.full(X.shape[1], fill_value, dtype=X.dtype)
+    
+        # Sample
+        elif strategy == "sample":
+            X = X.transpose()
+            mask = mask.transpose()
+
+            generators = np.empty(X.shape[0], dtype=object)
+
+            for i, (row, row_mask) in enumerate(zip(X[:], mask[:])):
+                row_mask = np.logical_not(row_mask).astype(np.bool)
+                row = row[row_mask]
+                if row.size > 0:
+                    uniques, counts = np.unique(row, return_counts=True)
+                    probas = counts / counts.sum()
+                    g = lambda k: random_state.choice(uniques, k, p=probas)
```

Comment:
We can't store a closure as an attribute. The estimator will not be picklable.

Reply:
I would rather just store the values and (if we go that way) their probabilities, as separate attributes.

Reply:
You mean 2 attributes for the values and their probas or separate from ``statistics_`` ?
I think making new attributes, different from ``statistics_`` would be confusing since they would only be used for that strategy.

Reply:
Yes. That's what I mean. Already, the attributes are fundamentally quite different from what's already there.